Revert "Pin minitest version to 5.21"

zzak opened the pull request:
Reverts rails/rails#50978

5.22.2 was released to address this, hopefully. :pray:

zzak committed:
Revert "Pin minitest version to 5.21"

zzak commented:
Still not fixed, so I've opened minitest/minitest#986 with a hack I wrote this morning.

Earlopain commented:
Is there any progress here, except for what I can read in the minitest PR? I'd hate for rails `7.1.4` to ship with the minitest version constraint.

skipkayhil committed:
Allow test command invocations to exit failure

Minitest will now exit 1 when a `-n` type test filter is used and no
test cases end up being run. However, these tests are not really
concerned with the exit code, just that `test:prepare` doesn't run.

This commit adds `allow_failure: true` to the test command invocations
so that we can continue to test whether `test:prepare` is run without
caring about the exit status of the test command.
